transformer/cnbcontainerizer.go

```diff
@@ -0,0 +1,174 @@
+			CNBBuilder: t.CNBConfig.BuilderImageName,
+		}}}}
+	return map[string][]transformertypes.Artifact{"": detectedServices}, nil
+}
+
+// Transform transforms the artifacts
+func (t *CNBContainerizer) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) (tPathMappings []transformertypes.PathMapping, tArtifacts []transformertypes.Artifact, err error) {
+	tArtifacts = []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		var cConfig artifacts.CNBMetadataConfig
+		err = a.GetConfig(artifacts.CNBMetadataConfigType, &cConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", cConfig, err)
+			continue
+		}
+		if cConfig.ImageName == "" {
+			cConfig.ImageName = common.MakeStringContainerImageNameCompliant(sConfig.ServiceName)
+		}
+		a.Configs[artifacts.CNBMetadataConfigType] = cConfig
+		ir := irtypes.NewIR()
+		ir.Name = t.Env.GetProjectName()
+		if prevIR, ok := a.Configs[irtypes.IRConfigType].(irtypes.IR); ok {
+			ir = prevIR
+		}
+		// Update an existing service with default port
+		if s, ok := ir.Services[sConfig.ServiceName]; ok {
+			if len(s.Containers) > 0 && len(s.Containers[0].Ports) == 0 {
+				// Add the port to the k8s pod.
+				s.Containers[0].Ports = []core.ContainerPort{{ContainerPort: common.DefaultServicePort}}
+				// Forward the port on the k8s service to the k8s pod.
+				port := networking.ServiceBackendPort{Number: common.DefaultServicePort}
+				s.AddPortForwarding(port, port, "")
+				ir.Services[sConfig.ServiceName] = s
+			}
+		} else {
+			// Create a new container in a new service in the IR (new or existing)
+			container := irtypes.NewContainer()
+			container.AddExposedPort(common.DefaultServicePort)
+			ir.AddContainer(cConfig.ImageName, container)
+			irService := irtypes.NewServiceWithName(sConfig.ServiceName)
+			serviceContainerPorts := []core.ContainerPort{}
+			for _, eport := range container.ExposedPorts {
+				// Add the port to the k8s pod.
+				serviceContainerPorts = append(serviceContainerPorts, core.ContainerPort{ContainerPort: eport})
+				// Forward the port on the k8s service to the k8s pod.
+				port := networking.ServiceBackendPort{Number: eport}
+				irService.AddPortForwarding(port, port, "")
+			}
+			serviceContainer := core.Container{Name: sConfig.ServiceName,
+				Image: cConfig.ImageName,
+				Ports: serviceContainerPorts}
+			irService.Containers = []core.Container{serviceContainer}
+			ir.Services[sConfig.ServiceName] = irService
+		}
+		a.Configs[irtypes.IRConfigType] = ir
+		tArtifacts = append(tArtifacts, transformertypes.Artifact{
+			Name:    a.Name,
+			Type:    artifacts.CNBMetadataArtifactType,
+			Paths:   a.Paths,
+			Configs: a.Configs,
+		})
```

Comment:
Lets split it into the two types we had discussed before. similar to https://github.com/konveyor/move2kube/blob/d31523bde478955c3c790c4bb6c4a17156ee73b5/transformer/dockerfilegenerator/golangdockerfiletransformer.go#L173 and https://github.com/konveyor/move2kube/blob/d31523bde478955c3c790c4bb6c4a17156ee73b5/transformer/dockerfilegenerator/golangdockerfiletransformer.go#L182

Reply:
We just have IR in the config. Nothing else. I have retained only one artifact type as before.